File: c++/operators/operators_logical.cpp
#include <iostream>

int main()
{
  bool a = true, b = false, c;

  std::cout << std::boolalpha;

  c = a && b;
  std::cout << a << " && " << b << " = " << c << std::endl;
  c = a && a;
  std::cout << a << " && " << a << " = " << c << std::endl;
  c = b && b;
  std::cout << b << " && " << b << " = " << c << std::endl;
  c = a || b;
  std::cout << a << " || " << b << " = " << c << std::endl;
  c = a || a;
  std::cout << a << " || " << a << " = " << c << std::endl;
  c = b || b;
  std::cout << b << " || " << b << " = " << c << std::endl;

  c = !a;
  std::cout << "!" << a << " = " << c << std::endl;
  c = !b;
  std::cout << "!" << b << " = " << c << std::endl;

  std::cout << std::endl;

  c = a and b;
  std::cout << a << " and " << b << " = " << c << std::endl;
  c = a and a;
  std::cout << a << " and " << a << " = " << c << std::endl;
  c = b and b;
  std::cout << b << " and " << b << " = " << c << std::endl;
  c = a or b;
  std::cout << a << " or " << b << " = " << c << std::endl;
  c = a or a;
  std::cout << a << " or " << a << " = " << c << std::endl;
  c = b or b;
  std::cout << b << " or " << b << " = " << c << std::endl;
  c = a xor b;
  std::cout << a << " xor " << b << " = " << c << std::endl;
  c = a xor a;
  std::cout << a << " xor " << a << " = " << c << std::endl;
  c = b xor b;
  std::cout << b << " xor " << b << " = " << c << std::endl;

  c = not a;
  std::cout << "not " << a << " = " << c << std::endl;
  c = not b;
  std::cout << "not " << b << " = " << c << std::endl;

  std::cout << std::boolalpha;
}

File: pascal/operators/operators_logical.pas
program OperatorsLogical;

var
  a: boolean = true;
  b: boolean = false;
  c: boolean;

begin
  c := a and b;
  writeln(a, ' and ', b, ' = ', c);
  c := a and a;
  writeln(a, ' and ', a, ' = ', c);
  c := b and b;
  writeln(b, ' and ', b, ' = ', c);
  c := a or b;
  writeln(a, ' or ', b, ' = ', c);
  c := a or a;
  writeln(a, ' or ', a, ' = ', c);
  c := b or b;
  writeln(b, ' or ', b, ' = ', c);
  c := a xor b;
  writeln(a, ' xor ', b, ' = ', c);
  c := a xor a;
  writeln(a, ' xor ', a, ' = ', c);
  c := b xor b;
  writeln(b, ' xor ', b, ' = ', c);

  c := not a;
  writeln('not ', a, ' = ', c);
  c := not b;
  writeln('not ', b, ' = ', c);
end.
